src/segment-loader.js

```diff
@@ -47,6 +47,50 @@ const detectEndOfStream = function(playlist, mediaSource, segmentIndex) {
 
 const finite = (num) => typeof num === 'number' && isFinite(num);
 
+export const illegalMediaSwitch = (loaderType, startingMedia, newSegmentMedia) => {
+  // Although these checks should most likely cover non 'main' types, for now it narrows
+  // the scope of our checks.
+  if (loaderType !== 'main' || !startingMedia || !newSegmentMedia) {
+    return;
+  }
+
+  if (!newSegmentMedia.containsAudio && !newSegmentMedia.containsVideo) {
+    return 'Neither audio nor video found in segment.';
+  }
+
+  if (startingMedia.containsAudio && startingMedia.containsVideo &&
+      newSegmentMedia.containsAudio && !newSegmentMedia.containsVideo) {
+    return 'Only audio found in segment when we expected audio and video.' +
+      ' We can\'t switch to audio only from audio and video.' +
+      ' To get rid of this message, please add codec information to the manifest.';
+  }
+
+  if (!startingMedia.containsAudio && startingMedia.containsVideo &&
+      newSegmentMedia.containsAudio && !newSegmentMedia.containsVideo) {
+    return 'Only audio found in segment when we expected video only.' +
```

Comment:
a bit misleading saying we expected "video only" as we can go from starting with video only to video + audio (sort of, if starting was video only because there was a separate audio segment loader and not because there was just only video stream)